src/Network/GRPC/MQTT/Client.hs

```diff
@@ -126,109 +156,109 @@ mqttRequest ::
   MQTTRequest streamtype request response ->
   IO (MQTTResult streamtype response)
 mqttRequest MQTTGRPCClient{..} baseTopic (MethodName method) useBatchedStream request = do
-  logDebug mqttLogger $ "Making gRPC request for method: " <> decodeUtf8 method
+  logDebug mqttLogger $ "Making gRPC request for method: " <> Text.Encoding.decodeUtf8 method
 
   handle handleMQTTException $ do
     sessionId <- generateSessionId rng
     let responseTopic = baseTopic <> "grpc" <> "session" <> sessionId
     let controlTopic = responseTopic <> "control"
     let baseRequestTopic = baseTopic <> "grpc" <> "request" <> sessionId
 
-    requestTopic <-
-      mkTopic (unTopic baseRequestTopic <> decodeUtf8 method)
-        `whenNothing` throw (MQTTException $ "gRPC method forms invalid topic: " <> decodeUtf8 method)
+    requestTopic <- case mkTopic (unTopic baseRequestTopic <> Text.Encoding.decodeUtf8 method) of
```

Comment:
The `case` reformulation is fine to me but I also thought the `whenNothing` form was nice and easy to read.